### PR TITLE
Generate seed csv

### PR DESCRIPTION
**Pull request by anacollado**

### CODE REVIEW

File: database/seed.js [hunk 1]
```diff
@@ -1,103 +1,108 @@
...
-    });
-};
+// const connection = mysql.createConnection({
+//   user: 'root',
+//   password: 'dobby',
+//   host: 'localhost'
+// });
 
-// seed database!
+// const db = Promise.promisifyAll(connection, { multiArgs: true });
```
Comment (bkwon94) [inline database/seed.js:18]: I would recommend cleaning up the commented code if it gets too long.

File: database/seedMethods.js [hunk 2]
```diff
@@ -2,14 +2,15 @@
 /* eslint-disable no-shadow */
 
 const Promise = require('bluebird');
```
Comment (bkwon94) [inline database/seedMethods.js:4]: Have you looked into async await? Highly recommend

File: database/seedMethods.js [hunk 3]
```diff
@@ -134,22 +135,58 @@ const seedDb = (data, db) => {
...
+  console.time(LABEL);
+  //use '|' delimiter to avoid conflict with fields with commas
+  const csvWriteStream = fastCSV.createWriteStream({headers: true, delimiter: '|'});
+  const fsWriteStream = fs.createWriteStream("bookDetails.csv");
+  fsWriteStream.on("finish", () => {
+    console.log("finished writing bookDetails CSV");
+    console.timeEnd(LABEL);
+  });
+  csvWriteStream.pipe(fsWriteStream);
+  let i = -1;
```
Comment (bkwon94) [inline database/seedMethods.js:149]: A little confused with this, maybe a comment could help.

File: database/seedMethods.js [hunk 4]
```diff
@@ -134,22 +135,58 @@ const seedDb = (data, db) => {
...
+  const writeNextRecordToCSV = () => {
+    i++;
+    if (i === size) {
+      return csvWriteStream.end();
+    }
+    let nextBook = createBook();
+    let canContinue = csvWriteStream.write(nextBook);
```
Comment (bkwon94) [inline database/seedMethods.js:156]: Smart!

**Review by bkwon94 — APPROVED** [review]

File: database/seedMethods.js (hunk 2 above)
Comment (rikotacards) [reply database/seedMethods.js:4]: +1 on Brian's comment. Though I recall using bluebird was how Andrew wrote the code. If everything is working fine and to db query time requirements then perhaps leave it for now.

File: database/index.js [hunk 5]
```diff
@@ -4,8 +4,9 @@ const Promise = require('bluebird');
 
 const connection = mysql.createConnection({
   user: 'root',
+  password:'dobby',
```
Comment (rikotacards) [inline database/index.js:7]: If the pw isn't specific to any personal pws then I guess its okay to leave it in here, but would be good practice to .gitignore any sensitive info.

File: database/seedMethods.js [hunk 6]
```diff
@@ -2,14 +2,41 @@
...
+const fs = require('fs');
+const fastCSV = require('fast-csv');
 
-// creates a data array for 100 BOOKS
 
-const createDataArray = () => {
+
+// function writeBooks() {
+//   while (i > 0) {
+//     if (!csvWriteStream.write(generateBooks())) {
```
Comment (rikotacards) [inline database/seedMethods.js]: Looks good here. But you may need to alter your generateBooks() function later to pass in the index i to be used as an ID in your noSQL db later for querying a specific book ID.

**Review by rikotacards — APPROVED** [review]

**Pull request merged**